tests/TestCase/Collection/CollectionTest.php

```diff
@@ -2632,4 +2632,21 @@ public function testArrayIteratorExtend()
         $this->assertTrue($newIterator->checkValues());
         $this->assertCount(3, $newIterator->toArray());
     }
+
+    /**
+     * Tests that elements in a lazy collection are only fetched once
```

Comment:
I think this title is a bit misleading.  While implicitly true, we deliberately test that the the filter callable will not be called for the lazy collection without exporting to array for example.

Shouldn't we add a test that verifies the actual expected behavior mention in this title?

Reply:
The test is actually verifying said behaviour. If the collection wasn't lazy, then the filters would e called immediately

Reply:
I agree. It's just a wording thing, I guess.
How about rewording to: "Tests that elements in a lazy collection are not fetched immediately"?

Reply:
sounds good to me